### PR TITLE
[Drop-In UI] Define colors and styles for day and night mode

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Fixes https://github.com/mapbox/navigation-sdks/issues/1440

Note: 
This PR only does the following:
- Define colors for day and night mode that the Drop-In UI will use
- Define styles for all the views supported by Drop-In UI
- Add code to `DropInNavigationView` to listen to theme changes
- Add code to the example activity to demonstrate changing between day and night mode.
- Disables maps scalebar
- Disables maps compass

This PR does not do the following:
- There is no code to read `libnavui-dropin/styles.xml#MapboxStyleNavigationView` and initialize attributes in Drop-In UI using typed array
- There is no code to apply style changes to the views at runtime.

### Screenshots or Gifs

https://user-images.githubusercontent.com/9770186/159137474-89fdea5e-1536-440f-bed2-ae254f92ca54.mp4

### CODE REVIEW

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt [hunk 1]
```diff
@@ -104,8 +107,15 @@ class DropInNavigationView @JvmOverloads constructor(
     }
 
     init {
+        val style = if (isNightModeEnabled()) {
+            NavigationStyles.NAVIGATION_NIGHT_STYLE
+        } else {
+            NavigationStyles.NAVIGATION_DAY_STYLE
+        }
+        binding.mapView.compass.enabled = false
+        binding.mapView.scalebar.enabled = false
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt:116]: Would like to move all of this away from the DropInNavigationView. 

Can this be moved into  `MapCoordinator.onAttached`.. maybe even `MapCoordinator.init`

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/MapboxNavigationViewCustomizedActivity.kt [hunk 2]
```diff
@@ -69,9 +72,21 @@ class MapboxNavigationViewCustomizedActivity : AppCompatActivity() {
             showCustomViews.value = showCustomViews.value?.not()
         }
 
+        binding.dayMode.setOnClickListener {
+            toggleTheme(AppCompatDelegate.MODE_NIGHT_NO)
+        }
+
+        binding.nightMode.setOnClickListener {
+            toggleTheme(AppCompatDelegate.MODE_NIGHT_YES)
+        }
```
Comment (kmadsen) [inline qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/MapboxNavigationViewCustomizedActivity.kt:81]: nit: Would be nice to have a single button switch here

File: libnavui-dropin/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -1,85 +1,192 @@
...
     </style>
 
-    <!-- Text -->
-
-    <style name="MapboxStyle.DropIn.TextAppearance.Regular" parent="">
-        <item name="android:textColor">@color/colorOnSurface</item>
-        <item name="android:textStyle">normal</item>
+    <!-- Audio guidance button -->
+    <style name="DropInStyleAudioGuidanceButton" parent="MapboxStyleExtendableButton">
+        <item name="extendableButtonIcon">@drawable/mapbox_ic_sound_on</item>
```
Comment (kmadsen) [inline libnavui-dropin/src/main/res/values/styles.xml:155]: Is there a style for `mapbox_ic_sound_off` 🤔

File: libnavui-dropin/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -1,85 +1,192 @@
...
+        <item name="android:layout_height">5dp</item>
+        <item name="android:layout_marginTop">10dp</item>
+        <item name="android:background">@drawable/mapbox_bg_handle</item>
+    </style>
+
+    <!-- Info panel styling -->
+    <style name="DropInInfoPanelHeadlineTextAppearance" parent="TextAppearance.AppCompat">
+        <item name="android:textColor">@color/colorOnSurface</item>
+        <item name="android:textStyle">normal</item>
+        <item name="android:textSize">27sp</item>
```
Comment (kmadsen) [inline libnavui-dropin/src/main/res/values/styles.xml:127]: nit: seems like this should reference another dimension

e.g., @dimen/mapbox_text_size_headline

Comment (kmadsen) [reply libnavui-dropin/src/main/res/values/styles.xml:127]: similar to other textSize styles

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/MapboxNavigationViewCustomizedActivity.kt (hunk 2 above)
Comment (kmadsen) [reply qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/MapboxNavigationViewCustomizedActivity.kt:81]: Planning on integrating an actual switch. so a single button works with text "Toggle night mode" https://github.com/mapbox/mapbox-navigation-android/pull/5594

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt (hunk 1 above)
Comment (abhishek1508) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt]: Done.

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/MapboxNavigationViewCustomizedActivity.kt (hunk 2 above)
Comment (abhishek1508) [reply qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/MapboxNavigationViewCustomizedActivity.kt]: Done.

File: libnavui-dropin/src/main/res/values/styles.xml (hunk 3 above)
Comment (abhishek1508) [reply libnavui-dropin/src/main/res/values/styles.xml:156]: Great catch.

File: libnavui-dropin/src/main/res/values/styles.xml (hunk 4 above)
Comment (abhishek1508) [reply libnavui-dropin/src/main/res/values/styles.xml]: Done.

File: libnavui-dropin/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -1,85 +1,197 @@
...
+    <style name="DropInSpeedLimitTextAppearance" parent="TextAppearance.AppCompat">
+        <item name="android:textColor">@color/colorOnSurface</item>
+        <item name="android:textSize">@dimen/mapbox_road_name_text_regular</item>
     </style>
 
-    <!-- Accessory -->
+    <!-- Road name view styling -->
+    <style name="DropInRoadNameViewTextAppearance" parent="TextAppearance.AppCompat">
+        <item name="android:textColor">@color/colorOnSurface</item>
+        <item name="android:textSize">14sp</item>
```
Comment (kmadsen) [inline libnavui-dropin/src/main/res/values/styles.xml:113]: I think this is supposed to be the road name

```suggestion
        <item name="android:textSize">@dimen/mapbox_road_name_text_regular</item>
```

File: libnavui-dropin/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -1,85 +1,197 @@
...
+        <item name="speedLimitBackgroundColor">@color/colorSurface</item>
+        <item name="speedLimitViennaBorderColor">@color/mapbox_speed_limit_view_vienna_border</item>
     </style>
 
-    <style name="MapboxStyle.DropIn.InfoPanelSubtitle" parent="">
-        <item name="android:textAppearance">@style/MapboxStyle.DropIn.TextAppearance.Regular</item>
-        <item name="android:textSize">15sp</item>
+    <style name="DropInSpeedLimitTextAppearance" parent="TextAppearance.AppCompat">
+        <item name="android:textColor">@color/colorOnSurface</item>
+        <item name="android:textSize">@dimen/mapbox_road_name_text_regular</item>
```
Comment (kmadsen) [inline libnavui-dropin/src/main/res/values/styles.xml:107]:
```suggestion
        <item name="android:textSize">@dimen/mapbox_speed_limit_text_regular</item>
```

**Pull request merged**